### PR TITLE
fix(pyspark): make sure `ibis.connect` works with pyspark

### PR DESCRIPTION
**Pull request by cpcloud**
Fixes a bug so that ibis.connect can be used with pyspark.

### CODE REVIEW

File: ibis/backends/pyspark/__init__.py [hunk 1]
```diff
@@ -101,12 +100,15 @@ class Options(ibis.config.BaseModel):
...
         if database := url.database:
-            params.append("spark.sql.warehouse.dir")
-            params.append(str(Path(database).absolute()))
+            params.append(
+                ("spark.sql.warehouse.dir", str(Path(database).absolute()))
+            )
 
-        builder = SparkSession.builder.config(*params)
+        builder = SparkSession.builder
+        for key, value in params:
```
Comment (kszucs) [inline ibis/backends/pyspark/__init__.py:110]: How about directly calling `builder.config()` instead of having an intermediate params container?

**Review by kszucs — APPROVED** [review]
Just a nit, otherwise it looks good to me.

File: ibis/backends/pyspark/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/pyspark/__init__.py:110]: You can't do that, `.config` takes a single key value pair. It's meant to be chained.

Comment (cpcloud) [reply ibis/backends/pyspark/__init__.py:110]: Ah, but I can use `SparkConf.setAll`!

**Pull request merged**